### DISCUSSION TITLE
Does SWC Respect an Initial Source Map?

### DISCUSSION THREAD

**Discussion by stahlbauer** [swc-project/swc#6108] [Q&A] [answered]
SWC can be applied to **files that already have a source map** (either inlined or as an external file). Is this mapping information **maintained** throughout all transformation passes of chained plugins?

I hope nobody is offended too much by all the questions regarding source maps :-D Thanks in advance.

**Comment by stahlbauer** [comment]
Minimizing a file that contained an initial source map with the swc config fragment
```
   "minify": {
      "compress": {
        "unused": true
      },
      "mangle": true
    }
```
results in a minimized version with an empty source map, that is, with 
```
//#sourceMappingURL=data:application/json;charset=utf-8;base64,IntcInZlcnNpb25cIjozLFwic291cmNlc1wiOltdLFwibmFtZXNcIjpbXSxcIm1hcHBpbmdzXCI6XCJcIixcImZpbGVcIjpcInN0ZG91dFwifSI=
```
which can be decoded to
```
"{\"version\":3,\"sources\":[],\"names\":[],\"mappings\":\"\",\"file\":\"stdout\"}"
```

**Reply by stahlbauer** [reply]
Passing the argument `--source-maps` results in a non-empty source map.

**Reply by stahlbauer** [reply]
(but at least the https://evanw.github.io/source-map-visualization/ cannot parse and visualize it)

**Comment by stahlbauer** [comment]
To contribute, it would be helpful for me to know where I can find unit/integration/system tests that test this functionality in the code base.

**Comment by jridgewell** [comment, marked as answer]
Using [this](https://play.swc.rs/?version=1.3.6&code=H4sIAAAAAAAAA2WQUa%2BaMBiG7%2F0VJLvZkiWCqCfMeCEoWo5oiqjIHS2cpg5aZgFly%2F77PmTnZMmaNF%2F7fk%2Fftu9bLWjFpdDepPz8Rfs10GDcsqq%2BCc2YDX4Pskcpb5VGpVCV9tDmPTgbDIefNCXrG838pCy5YMdgO0%2BTKvkG25zTpHMdXpUUM5KobDr%2Bmt0lQ443iiPvZ3K2anSVzG%2Fvr8hZ8LjIVXxAU%2BiXRNhG6kx%2B0BYphy8YErsmjbxrfPIaYmK%2B5wuOeH%2BOjiyDFrsc2ClyVaexfzx0ajLeeUSHnieFW8WRDx42QUXckBbxj97ZvdN1Xsem%2F7wDH10bHx4h1BDqBupmr%2Bc2NpAKddcO28cStOV2tbPx0X%2Fp1hfea3voYx2rEDzCwztnAKcr3PXa3m%2B7SkGjKgAu%2BMv99zd91xAR5ETg57ufmXSZiVN9MYOSjMYsLqzm1fk%2BReuJQc5eTrn9Eq3HHcdocdIhvxotV6BNLJh5trEbKjC7jKyamphlzsJC6y4PVu5bBPfbqSP0%2BR%2FUVtyeHwIAAA%3D%3D&config=H4sIAAAAAAAAA22TTZLjIAyF78I625lFDjC7PgNFjHDUg5ELiXRcqdx9hH9wnMwOPp6EeBIP882dOT%2FM6DJDriuekri7ORuZRuAu4yjmZIQVSS7w1LXLPYgqgH%2FpUSRiMOfgIsPJDJgwTDWR0BjhBnGJO5mOhjEDcz3TDGWAJNziXM70w5v2QhTBpY%2B9dWwxCfSQ99COYnQjg725zK%2B3uYxM6YCKgLdjpvGFJo%2BClDTZxjw4bzvy0ABm6ARvsEs0h0oSa2Gtkhl6uJS%2Br2YuSri5WJy0XHCfbdAbW9yVkMWGkvgNHSpd0PLIVYXBZpCS06b5JkwHH%2F4CaJ3RMSc3wB4586B9OCrDf1SYgjZVpka143tVCXp9nEUM7YW1asiCu1sZfOmgvrDbk6%2Fw8CBGDxZCULdbMP%2BgdNc9WR1MCm2rrrmw92DZ2m2ejrSOwAf8o1XJaztWPji5vjOehgvFjxQDyJX8B1YjhN5h1um9j%2B%2B0JA9qM%2FiXg8J128ZUmy9kY%2F16m886EBpj%2B0iXNr76QweX%2BjaWzwrIlwoes3nz1%2F1tnvtvXfIhf6261XgquYMvVy0zmKJeZZ7%2FAPQlah0zBAAA) (which I generated from [TS playground](https://www.typescriptlang.org/play?inlineSources=true&stripInternal=false&inlineSourceMap=true#code/GYVwdgxgLglg9mABMOcAUBKAXIsIC2ARgKYBOiA3gFCK2KnFQilICMA3FQL5XEAeABzikoiCAgDOovogC8yVJnZA)), it seems the input sourcemap _is_ maintained throughout the plugins. 


It seems it does not propagate the `sourcesContent` field (instead, it uses the content input into SWC).
